numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -3452,7 +3455,24 @@ def device_memset(dst, val, size, stream=0):
     else:
         fn = driver.cuMemsetD8
 
-    fn(device_pointer(dst), val, size, *varargs)
+    try:
+        fn(ptr, val, size, *varargs)
+    except CudaAPIError as e:
+        invalid = (
```

Comment:
How do you create a situation which this code is intended to handle? Is there an example that could be turned into a test?

Reply:
The test suite was already catching it.  I'm was running into unit test failures running on WSL.  One was lack of support for IPC and the other was memset throwing errors when attempting to access memory accessible by host and device.  This fallback enables the test suite to complete successfully.